contracts/contracts/ramps/paylah/PaylahRamp.sol

```diff
@@ -235,20 +238,22 @@ contract PaylahRamp is Ownable {
     /**
      * @notice Generates a deposit entry for off-rampers that can then be fulfilled by an on-ramper. This function will not add to
      * previous deposits. Every deposit has it's own unique identifier. User must approve the contract to transfer the deposit amount
-     * of USDC.
+     * of USDC. In order to create a deposit the depositor must submit an offRampIdHash that has been linked to a valid payment link
+     * and added via governance. If the passed _offRampIdHash is not linked to a paymentLink then the function will revert.
      *
-     * @param _upiId            The packed upi ID of the depositor
+     * @param _offRampIdHash    Depositor's off-ramp id hash which has been linked to a valid payment link and validated by governance
      * @param _depositAmount    The amount of USDC to off-ramp
      * @param _receiveAmount    The amount of USD to receive
      */
     function offRamp(
-        uint256[8] memory _upiId,
+        bytes32 _offRampIdHash,
```

Comment:
When in the UI are you thinking the offramp could provide their payment link? I was thinking during creating the deposit. We can store it in the `Deposit` struct.

Reply:
Feels like they should just provide it to us. We validate it and then add it on-chain as valid. Don't think they should need to re-enter into the app at all. They would need to enter their alias and phone number tho